great_expectations/execution_engine/redshift_execution_engine.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import annotations
+
+import re
+from typing import Any, Optional, cast
+
+from great_expectations.compatibility import aws
+from great_expectations.compatibility.sqlalchemy import sqlalchemy as sa
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.exceptions.exceptions import GreatExpectationsError
+from great_expectations.execution_engine.sqlalchemy_batch_data import SqlAlchemyBatchData
+from great_expectations.execution_engine.sqlalchemy_execution_engine import (
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.metric_provider import metric_value
+from great_expectations.expectations.metrics.table_metrics.table_column_types import (
+    ColumnTypes as BaseColumnTypes,
+)
+
+
+class RedshiftExecutionEngine(SqlAlchemyExecutionEngine):
```

Comment:
I add a redshift specific execution engine so I can write metrics specific to redshift and isolate this from the rest of our sqlalchemy datasources. This is to prevent conditional logic in our metrics code that switches based on dialect.